sdk/azcore/runtime/policy_http_trace.go

```diff
@@ -0,0 +1,117 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+)
+
+const (
+	attrHTTPMethod     = "http.method"
+	attrHTTPURL        = "http.url"
+	attrHTTPUserAgent  = "http.user_agent"
+	attrHTTPStatusCode = "http.status_code"
+
+	attrAZClientReqID  = "az.client_request_id"
+	attrAZServiceReqID = "az.service_request_id"
+
+	attrNetPeerName = "net.peer.name"
+)
+
+// newHTTPTracePolicy creates a new instance of the httpTracePolicy.
+//   - allowedQueryParams contains the user-specified query parameters that don't need to be redacted from the trace
+func newHTTPTracePolicy(allowedQueryParams []string) exported.Policy {
```

Comment:
What's the rationale for not exporting this?

Reply:
Since the policy is always added to the pipeline and is a no-op if tracing isn't enabled, there's no need to export it for a custom pipeline.

This highlights a bit of a design flaw.  Originally, we let callers create an empty pipeline so they could add their own policies piecemeal; this required policy constructors to be exported.  We later removed the ability to create an empty pipeline but forgot to unexport all of the policy constructors.

I think the rule should be that if a policy is always added to the pipeline, there's no need to export its constructor.  IFF the policy is not in the pipeline by default should its constructor be exported (seems unlikely that we'd create such a policy, but I suppose it's possible).

@JeffreyRichter what are your thoughts on this?